cgm_diabetes/captioning/generate_captions.py

```
@@ -0,0 +1,502 @@
+            processed += 1
+
+            # Save after every patient so no progress is ever lost
+            save_captions(captions, output_path)
+            print(f"    ✓ Saved ({len(captions)} total)")
+
+        except Exception as e:
+            print(f"API failed for {patient_id}: {e}")
+            failed.append(patient_id)
+
+    # Summary
+    print(f"\n[captions] Done.")
```

Ruff (0.15.6)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)